staging/src/k8s.io/apiserver/pkg/server/handler.go

```diff
@@ -92,12 +114,57 @@ func (a *APIServerHandler) ListedPaths() []string {
 	for _, ws := range a.GoRestfulContainer.RegisteredWebServices() {
 		handledPaths = append(handledPaths, ws.RootPath())
 	}
-	handledPaths = append(handledPaths, a.PostGoRestfulMux.ListedPaths()...)
+	handledPaths = append(handledPaths, a.NonGoRestfulMux.ListedPaths()...)
 	sort.Strings(handledPaths)
 
 	return handledPaths
 }
 
+type director struct {
+	name               string
+	goRestfulContainer *restful.Container
+	nonGoRestfulMux    *mux.PathRecorderMux
+}
+
+func (d director) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	path := req.URL.Path
+
+	// if we are exactly /apis or /apis/, pass then we need special handling in loop.
+	// normally these are passed to the nonGoRestfulMux, but if discovery is enabled, it will go directly.
+	// We can't rely on a prefix match since /apis matches everything (see the big comment on Director above)
+	isDiscoveryRequest := path == "/apis" || path == "/apis/"
```

Comment:
move inside the `ws.RootPath() == "/apis"` case? we'll only have one webservice registered with that root path, right?